### DISCUSSION TITLE
q-expansion-item root width and retrieving child width issues

### DISCUSSION THREAD

**Discussion by amirny2205** [quasarframework/quasar#15309] [General - Components / Directives / etc]
Hello!
I want to make q-expansion-item root element(the one with arrow on my screenshots) be as small as it's content
Though I have faced the following difficulties:

1) inner q-list inherits style when unfolding, thus taking up the same width. It takes normal style when 100% unfolded
![q-expansion-item root width issue 1](https://user-images.githubusercontent.com/105064582/213842850-382e72f5-cd5a-4c84-8dc6-e60f4689084d.png)
![q-expansion-item root width issue 2](https://user-images.githubusercontent.com/105064582/213842877-9609d885-ab00-470c-9259-36c48e30ba34.png)
![q-expansion-item root width issue 3](https://user-images.githubusercontent.com/105064582/213842887-a26dd0d2-982c-46c4-90de-a9942d904208.png)
2) width of the child q-list @click equals to 0
I need to reposition the q-list to be centered under the expansion-item root.
So I need to take it's width and put it in position css attributes ("left" or "right")
But when I try to retrieve it, it equals to 0.
What am I doing \ understanding wrong? Are there any other ways to achieve the same result? q-menu doesn't have "expansion" effect

## codepen
[codepen](https://codepen.io/amirny2205/pen/GRBydQz)

## code

plese notice the console.log
```
<template>
  <q-expansion-item
    expand-separator
    hide-expand-icon
    style="margin: 16px; width: 20px; background-color: #688;"
    @click="retrieve_width"
  >
    <template v-slot:default>
    <q-list style="display: flex; flex-direction:column; position: absolute;" id="q-l">
      <q-btn>
	option 1
      </q-btn>
      <q-btn>
	option 2
      </q-btn>
      <q-btn>
	option 3
      </q-btn>
      <q-btn>
	option 4
      </q-btn>
    </q-list>
    </template>
    <template v-slot:header>
      <q-icon name="arrow_drop_down" style="margin-top:5px"/>
    </template>
  </q-expansion-item>
</template>
<script setup>
 function retrieve_width() {
   let el = document.getElementById('q-l')
   console.log(el.offsetWidth)
 }
</script>
<style>
 body {
     background-color: #866;
 }
 .q-expansion-item .q-item{
    padding:0;
    height: 35px;
    min-height: 35px;
}
</style>
```

**Comment by pdanpdan** [comment]
Why don't you use either a QBtnDropdown or use a QMenu on a button?

**Reply by amirny2205** [reply]
> Why don't you use either a QBtnDropdown or use a QMenu on a button?

Well expansion item has a different effect. QMenu is definetely a workaround but the effect is different. I chose not to hurry and ask here.
